modules/data-loading/pages/data-streaming-connector/big-query.adoc

```diff
@@ -0,0 +1,269 @@
+This parameter supports keys in .p12 or .json format.
+Required if you set `AuthValue` to `0`.
+|`OAuthAccessToken`
+|Access token of your Google Service Account.
+See Google Cloud documentation to https://cloud.google.com/iam/docs/create-short-lived-credentials-direct#sa-credentials-oauth[generate an access token].
+Required if you set `AuthValue` to `1`.
+|`OAuthRefreshToken`
+|Refresh token of your Google Service Account.
+See Google Cloud documentation to https://developers.google.com/identity/protocols/oauth2/web-server#exchange-authorization-code[generate a refresh token].
+Required if you set `AuthValue` to `1`.
+|===
+
+==== Value converter
+BigQuery stores data in a proprietary columnar format.
+In order for TigerGraph to ingest the data, Data Streaming Connector uses a value converter to convert the data into the CSV format.
+To do this, add a line in the configuration file that specifies the value converter:
+
+[source.wrap,text]
+value.converter=com.tigergraph.kafka.connect.converters.TigerGraphCsvConverter
+
+==== Subsection configurations
+Subsection configurations are specified after the overall connector configurations.
+Each set of subsection configurations creates an instance of a connector by specifying a connector name and a Kafka topic that the connector streams messages to.
+
+When streaming from BigQuery, each subsection also must have a SQL query statement used to query your BigQuery warehouse.
+The results of the query are streamed to the corresponding topic.
+
+NOTE: If you are querying `STRUCT` data or arrays, see
+
+For example, the following subsection configurations create two instances of the connector, each streaming the results of their corresponding queries to `topic_0` and `topic_1`:
+
+[source.wrap,text]
+----
+[bq_query_0]
+name="bq_query_0"
+topic="topic_0"
+query="SELECT * FROM `bigquery-public-data.breathe.arxiv` LIMIT 1000"
+
+[bq_query_1]
+name="bq_query_1"
+topic="topic_1"
+query="SELECT doi,abstract,authors FROM `bigquery-public-data.breathe.nature`"
+----
+
+include::partial$create-connector.adoc[
+
+include::partial$create-data-source.adoc[]
+
+include::partial$create-loading-job-kafka.adoc[]
+
+include::partial$run-loading-job.adoc[]
+
+== Querying `STRUCT` data and arrays
+If the record being queried in your BigQuery warehouse contains `STRUCT` data or arrays,  conversion functions need to be applied to the SQL statement:
+
+=== `STRUCT` data
+
+When retrieving struct data, it is recommended to retrieve the fields of the data directly.
+For example:
+
+[.wrap,sql]
+----
+SELECT basic.age, basic.gender FROM `project.dataset.table`
+----
+
+If you want to retrieve the entire `STRUCT`, you need to first use `TO_JSON_STRING()` to convert the `STRUCT` data into JSON strings in the converted CSV streams, and then xref:gsql-ref:ddl-and-loading:creating-a-loading-job.adoc#_flatten_a_json_object_in_a_csv_file[flatten the JSON strings] when loading the data into the graph.
```

Comment:
We'd better give an example like L152, since `TO_JSON_STRING()` is a BigQuery built-in function and users might don't know it. `SELECT TO_JSON_STRING(col) FROM table`